### PR TITLE
Group chats hi-fi styling

### PR DESCRIPTION
**Pull request by lucaslcode**
The first in many messaging pull requests

### CODE REVIEW

**Comment by lucaslcode** [comment]
This is already getting big so I'll stop it here for review.
Questions: Some things I'm not sure if I should be styling individual components where they are used or styling for everything.  I'll go through and add some comments.

File: app/frontend/src/components/Avatar/Avatar.tsx
Comment (lucaslcode) [inline app/frontend/src/components/Avatar/Avatar.tsx:47]: This seems hacky, but passing in a class with width and height was getting overridden by the default values.

File: app/frontend/src/features/messages/Messages.tsx
Comment (lucaslcode) [inline app/frontend/src/features/messages/Messages.tsx:69]: Is doing this to apply the tab bar and title only to some routes okay?

File: app/frontend/src/features/messages/groupchats/GroupChatList.tsx
Comment (lucaslcode) [inline app/frontend/src/features/messages/groupchats/GroupChatList.tsx]: Should this be applied more generally somewhere? The list items were getting an ugly default link colour and underline.

Comment (lucaslcode) [reply app/frontend/src/features/messages/groupchats/GroupChatList.tsx]: Maybe  `a > .MuiList-root : ` in the theme file?

File: app/frontend/src/features/messages/groupchats/GroupChatListItem.tsx
Comment (lucaslcode) [inline app/frontend/src/features/messages/groupchats/GroupChatListItem.tsx:66]: `react-query` is until until you get something like this...

File: app/frontend/src/features/userQueries/useUsers.ts
Comment (lucaslcode) [inline app/frontend/src/features/userQueries/useUsers.ts]: Had to make this accept undefined ids because now the chats get the chat id from the url, load the chat, and then use that to load the list of members. So before the group chat is loaded, the list of members is undefined.

File: app/frontend/src/theme.ts
Comment (lucaslcode) [inline app/frontend/src/theme.ts]: I actually got rid of the TabPanel in messages. But maybe I should leave this in for the connections page? The whitespace is too big, because there is already padding from the outermost container.

```suggestion
  },
```

**Comment by lucaslcode** [comment]
Noticed: Buttons next to message title should have outline, send button is stretched on desktop, add padding and negative margin to list items so the highlight has white space

File: app/frontend/src/components/Avatar/Avatar.tsx
Comment (darrenvong) [reply app/frontend/src/components/Avatar/Avatar.tsx:47]: Wouldn't putting `classes.defaultSize` before `className` in the list do the same thing? I assume JSS emits style in the order the classes are defined, but could be wrong on that.

File: app/frontend/src/App.tsx
Comment (darrenvong) [inline app/frontend/src/App.tsx:44]: Are we replacing this with a logo later?

File: app/frontend/src/components/Dialog.stories.tsx
Comment (darrenvong) [inline app/frontend/src/components/Dialog.stories.tsx]: Why do we need `@ts-ignore` here before the export?

File: app/frontend/src/features/messages/Messages.tsx
Comment (darrenvong) [reply app/frontend/src/features/messages/Messages.tsx:69]: Yeah I think it's ok in this case. I think something weird happens with this kinda code and `history.push` together only when used with event handlers from `react-hook-form` after I was digging around more with it last week, still not sure why tbh 🤷

File: app/frontend/src/features/messages/groupchats/CreateGroupChat.tsx
Comment (darrenvong) [inline app/frontend/src/features/messages/groupchats/CreateGroupChat.tsx]: I noticed that if there's an error, you close the dialog and open it again, the error is still there. Might it be better to clear it here when the user closes the dialog?

File: app/frontend/src/features/messages/groupchats/GroupChatListItem.tsx
Comment (darrenvong) [reply app/frontend/src/features/messages/groupchats/GroupChatListItem.tsx:66]: Lol I have to say the way the message type is structured feels a bit weird/awkward, although even with a tighter structure, I doubt we can avoid the switch logic of checking through message type. 

Since this is after the all the hooks usage (so there shouldn't be warnings of using hooks conditionally), wouldn't this be clearer if we write it out more explicitly rather than these crazy nesting of ternaries?

File: app/frontend/src/features/messages/groupchats/GroupChatView.tsx
Comment (darrenvong) [inline app/frontend/src/features/messages/groupchats/GroupChatView.tsx:47]: This goes to a blank chat if a user has anything non numerical in the URL 🤔

File: app/frontend/src/features/messages/groupchats/GroupChatsTab.tsx
Comment (darrenvong) [inline app/frontend/src/features/messages/groupchats/GroupChatsTab.tsx:20]: It looks fine for me without this set and it's creating an unnecessary horizontal scroll bar on tablet size width. What was this trying to fix at first?

File: app/frontend/src/features/messages/messagelist/MessageView.tsx
Comment (darrenvong) [inline app/frontend/src/features/messages/messagelist/MessageView.tsx]: Couldn't we do something similar to what you did on line 86 here and conditionally apply class names so all styles are in the `makeStyles(...)` for consistency?

File: app/frontend/src/features/messages/messagelist/MessageView.tsx
Comment (darrenvong) [inline app/frontend/src/features/messages/messagelist/MessageView.tsx]: Couldn't these styles live in `TimeInterval` instead, so it's closer to where it's needed?

File: app/frontend/src/features/messages/surfing/HostRequestListItem.tsx
Comment (darrenvong) [inline app/frontend/src/features/messages/surfing/HostRequestListItem.tsx:40]:
```suggestion
            {otherUser && <Typography className={classes.name}>{otherUser.name}</Typography>}
```
Probably still want to display the other user's name here?

File: app/frontend/src/features/userQueries/useUsers.ts
Comment (darrenvong) [reply app/frontend/src/features/userQueries/useUsers.ts]: Couldn't we have the group chat view default to an empty list being passed in here instead? Won't need to update the logic here at all that way and it's way less code?

File: app/frontend/src/theme.ts
Comment (darrenvong) [reply app/frontend/src/theme.ts]: Yeah I think the tiles are right underneath the tabs without this in the connections page. Although I'm thinking for consistency with how far the first message block is from the tabs (currently 8px or `theme.spacing(1)`) on the messages page, I think might be best to set it to that as default here instead?

**Review by darrenvong — COMMENTED** [review]
Loving the progress on this. This page is gonna look so much nicer once this is properly merged in!

File: app/frontend/src/components/Avatar/Avatar.tsx
Comment (lucaslcode) [reply app/frontend/src/components/Avatar/Avatar.tsx:47]: Had no effect.

File: app/frontend/src/features/messages/groupchats/GroupChatListItem.tsx
Comment (lucaslcode) [reply app/frontend/src/features/messages/groupchats/GroupChatListItem.tsx:66]: Any suggestions? I've tried making it clearer in the latest commit (will push soon) but the problem is needing the user's name in the control message text.

File: app/frontend/src/App.tsx
Comment (lucaslcode) [reply app/frontend/src/App.tsx:44]: Not sure, but there is no "site title" on mobile in the design

File: app/frontend/src/components/Dialog.stories.tsx
Comment (lucaslcode) [reply app/frontend/src/components/Dialog.stories.tsx]: leftover from something else

File: app/frontend/src/features/messages/groupchats/GroupChatsTab.tsx
Comment (lucaslcode) [reply app/frontend/src/features/messages/groupchats/GroupChatsTab.tsx:20]: It's so the darkening background extends out to the side a bit. I'll make overflow visible

File: app/frontend/src/features/messages/messagelist/MessageView.tsx
Comment (lucaslcode) [reply app/frontend/src/features/messages/messagelist/MessageView.tsx]: I think `TimeInterval` is just to abstract from moment, and might be used in all kinds of places?

Comment (lucaslcode) [reply app/frontend/src/features/messages/messagelist/MessageView.tsx]: Though it's in messages. I'll change it.

File: app/frontend/src/components/Avatar/Avatar.tsx
Comment (darrenvong) [reply app/frontend/src/components/Avatar/Avatar.tsx:47]: Hmm that's probably the best we can do then. I think this is better than relying on CSS definition order tbh, that feels more brittle to me.

File: app/frontend/src/features/messages/groupchats/GroupChatsTab.tsx
Comment (darrenvong) [reply app/frontend/src/features/messages/groupchats/GroupChatsTab.tsx:20]: Is this because of the fixed width container globally on the page? Wonder if we should review whether that's still a good idea at a later stage if that's the case, since I remember Aapeli mentioning wanting to maybe do a full window/page width map before.

File: app/frontend/src/features/messages/groupchats/GroupChatListItem.tsx
Comment (darrenvong) [inline app/frontend/src/features/messages/groupchats/GroupChatListItem.tsx]:
```typescript
const targetName = firstName(
  groupChatMembersQuery.data?.get(
    messageTargetId(groupChat.latestMessage)
  )?.name
) || ""
```
We could simplify to that instead since there's already a check for `groupChat.latestMessage` above so we know it's truthy here.

File: app/frontend/src/features/messages/groupchats/GroupChatListItem.tsx
Comment (darrenvong) [reply app/frontend/src/features/messages/groupchats/GroupChatListItem.tsx:66]: What if we lift up `authorName` outside of the if and merge the conditions together so we have something like:
```typescript
const authorName = ...;
if (groupChat.latestMessage && isControlMessage(groupChat.latestMessage)) {
  // do control message stuff
} else {
 text = `${authorName}: ${groupChat.latestMessage.text?.text || ""}`;
}
```
Combined with the simplification on `targetName` above, we've flatten it to one if-else so it's not too bad to read again. Could also shorten it by a few more characters here if we move the `|| ""` for defaulting to empty string logic inside the `firstName` helper?

File: app/frontend/src/features/messages/SendField.tsx
Comment (darrenvong) [inline app/frontend/src/features/messages/SendField.tsx:59]: I noticed when a message is sent, this input is not cleared. We should probably do that since that's quite standard chat behaviour.

File: app/frontend/src/features/messages/surfing/HostRequestListItem.tsx
Comment (darrenvong) [reply app/frontend/src/features/messages/surfing/HostRequestListItem.tsx:40]: Or is this not the case since you deleted the `UserName` component?

File: app/frontend/src/features/messages/groupchats/GroupChatView.tsx
Comment (darrenvong) [inline app/frontend/src/features/messages/groupchats/GroupChatView.tsx]: The sides of this popup on this is right against the text, so could do with some side padding?

**Review by darrenvong — CHANGES_REQUESTED** [review]
Almost there!

File: app/frontend/src/features/messages/SendField.tsx
Comment (darrenvong) [reply app/frontend/src/features/messages/SendField.tsx:59]: Wait is this fixed in the next PR in #513?

File: app/frontend/src/features/messages/surfing/HostRequestListItem.tsx
Comment (lucaslcode) [reply app/frontend/src/features/messages/surfing/HostRequestListItem.tsx:40]: I just left it for the next PR, it's done there

File: app/frontend/src/features/messages/SendField.tsx
Comment (lucaslcode) [reply app/frontend/src/features/messages/SendField.tsx:59]: yes

File: app/frontend/src/features/messages/groupchats/GroupChatView.tsx
Comment (lucaslcode) [reply app/frontend/src/features/messages/groupchats/GroupChatView.tsx]: Done, this is because of the override I put in theme.ts... 🤔

**Review by darrenvong — APPROVED** [review]

**Pull request merged**
